src/Components/Aspire.Azure.AI.OpenAI/ConfigurationSchema.json

```diff
@@ -69,12 +69,12 @@
                           "properties": {
                             "Delay": {
                               "type": "string",
-                              "format": "duration",
+                              "pattern": "^-?(\\d{1,7}|((\\d{1,7}[\\.:])?(([01]?\\d|2[0-3]):[0-5]?\\d|([01]?\\d|2[0-3]):[0-5]?\\d:[0-5]?\\d)(\\.\\d{1,7})?))$",
```

Comment:
did you author this based on the spec (appears to be BNF) or is it copied from someplace?

and assuming https://datatracker.ietf.org/doc/html/rfc3339#appendix-A is the right place to look at, what does this correspond to -- "Duration" here?

Mentally translating that BNF into pattern is a bit taxing on my brain. I see you have a bunch of tests -- are there any other canonical tests for this patter we can borrow from somewhere?

Reply:
Oh I see it's documented in the code below. But still not sure which part of the spec it matches to.

Reply:
> and assuming https://datatracker.ietf.org/doc/html/rfc3339#appendix-A is the right place to look at, what does this correspond to -- "Duration" here?

That is precisely the wrong place to look. We used to say `format: duration` which says:

> A duration as defined by the [ISO 8601 ABNF for "duration"](https://datatracker.ietf.org/doc/html/rfc3339#appendix-A). For example, P3D expresses a duration of 3 days.

But this doesn't work with `TimeSpan.Parse` which is what the code is using.

> did you author this based on the spec (appears to be BNF) or is it copied from someplace?

I authored it following the behavior of `TimeSpan.Parse`, which is what the code that is parsing the string uses. See: 

https://github.com/dotnet/runtime/blob/c19d68e52c6d743c8f81ed9049e8c172ae461c99/src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs#L302-L304

https://github.com/dotnet/runtime/blob/c19d68e52c6d743c8f81ed9049e8c172ae461c99/src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Emitter/CoreBindingHelpers.cs#L620-L623